packages/devtools_app/lib/src/flutter/split.dart

```diff
@@ -100,29 +101,70 @@ class _SplitState extends State<Split> {
-    firstSize = firstSize.clamp(halfDivider, axisSize - halfDivider);
-    secondSize = secondSize.clamp(halfDivider, axisSize - halfDivider);
+    final numDividers = widget.children.length - 1;
+    final sizes = List.generate(
+      fractions.length,
+      (i) =>
+          (axisSize - numDividers * Split.dividerMainAxisSize) * fractions[i],
+    );
 
-    // Remove space from each child to place the divider in the middle.
-    firstSize = firstSize - halfDivider;
-    secondSize = secondSize - halfDivider;
+    void updateSpacing(DragUpdateDetails dragDetails, int splitterIndex) {
+      final dragDelta =
+          isHorizontal ? dragDetails.delta.dx : dragDetails.delta.dy;
+      final fractionalDelta = dragDelta / axisSize;
+
+      double updateSpacingBeforeSplitterIndex(double delta) {
+        var index = splitterIndex;
+        while (index >= 0) {
+          fractions[index] += delta;
+          if (fractions[index] >= 0.0) {
+            _clampFraction(index);
+            return 0.0;
+          }
+          delta = fractions[index];
+          _clampFraction(index);
+          index--;
+        }
+        return delta;
+      }
+
+      double updateSpacingAfterSplitterIndex(double delta) {
+        var index = splitterIndex + 1;
+        while (index < fractions.length) {
+          fractions[index] += delta;
+          if (fractions[index] >= 0.0) {
+            _clampFraction(index);
+            return 0.0;
+          }
+          delta = fractions[index];
+          _clampFraction(index);
+          index++;
+        }
+        return delta;
+      }
 
-    void updateSpacing(DragUpdateDetails dragDetails) {
-      final delta = isHorizontal ? dragDetails.delta.dx : dragDetails.delta.dy;
-      final fractionalDelta = delta / axisSize;
       setState(() {
-        // Update the fraction of space consumed by the children,
-        // being sure not to allocate any negative space.
-        firstFraction += fractionalDelta;
-        firstFraction = firstFraction.clamp(0.0, 1.0);
+        // Update the fraction of space consumed by the children. Always update
+        // the shrinking children first so that we do not over-increase the size
+        // of the growing children and cause layout overflow errors.
+        if (fractionalDelta <= 0.0) {
+          final overflowDelta =
+              updateSpacingBeforeSplitterIndex(fractionalDelta);
+          final actualDelta = overflowDelta != 0.0
+              ? fractionalDelta - overflowDelta
+              : fractionalDelta;
+          updateSpacingAfterSplitterIndex(-actualDelta);
+        } else {
+          final overflowDelta =
+              updateSpacingAfterSplitterIndex(-fractionalDelta);
+          final actualDelta = overflowDelta != 0.0
+              ? fractionalDelta + overflowDelta
+              : fractionalDelta;
+          updateSpacingBeforeSplitterIndex(actualDelta);
+        }
       });
```

Comment:
this conditional isn't needed as both conditions are equivalent.

Reply:
removed